torchvision/prototype/transforms/_geometry.py

```diff
@@ -136,30 +136,39 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         )
 
 
-class MultiCropResult(list):
-    """Helper class for :class:`~torchvision.prototype.transforms.BatchMultiCrop`.
-
-    Outputs of multi crop transforms such as :class:`~torchvision.prototype.transforms.FiveCrop` and
-    `:class:`~torchvision.prototype.transforms.TenCrop` should be wrapped in this in order to be batched correctly by
-    :class:`~torchvision.prototype.transforms.BatchMultiCrop`.
+class FiveCrop(Transform):
+    """
+    Example:
+        >>> class BatchMultiCrop(transforms.Transform):
+        ...     def forward(self, sample: Tuple[List[features.Image], features.Label]):
+        ...         images, labels = sample
+        ...         batch_size = len(images)
+        ...         images = features.Image.new_like(images[0], torch.stack(images))
+        ...         labels = features.Label.new_like(labels, labels.repeat(batch_size, *[1] * labels.ndim))
+        ...         return images, labels
+        ...
+        >>> image = features.Image(torch.rand(3, 256, 256))
+        >>> label = features.Label(0)
+        >>> transform = transforms.Compose([transforms.FiveCrop(), BatchMultiCrop()])
+        >>> images, labels = transform(image, label)
+        >>> images.shape
+        torch.Size([5, 3, 224, 224])
+        >>> labels.shape
+        torch.Size([5])
     """
 
-    pass
-
-
-class FiveCrop(Transform):
     def __init__(self, size: Union[int, Sequence[int]]) -> None:
         super().__init__()
         self.size = _setup_size(size, error_msg="Please provide only two dimensions (h, w) for size.")
 
     def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
         if isinstance(inpt, features.Image):
             output = F.five_crop_image_tensor(inpt, self.size)
-            return MultiCropResult(features.Image.new_like(inpt, o) for o in output)
+            return [features.Image.new_like(inpt, o) for o in output]
         elif is_simple_tensor(inpt):
-            return MultiCropResult(F.five_crop_image_tensor(inpt, self.size))
+            return list(F.five_crop_image_tensor(inpt, self.size))
```

Comment:
`five_crop_*` currently returns a tuple whereas `ten_crop_*` returns a list. We could full BC here and also return a tuple, but I think later resolving https://github.com/pytorch/vision/issues/6455#issuecomment-1222114489 in a BC manner would be harder due to the different types. Since `MultiCropResult` already was a list, we somewhat agreed to this BC break before.

Reply:
Could you please add a comment to highlight that this is potentially BC-breaking and worth revising before releasing it?

Reply:
Should we maybe have this change on `F.five_crop` rather than on the transform? If we accept this potential BC break on the transform we should accept the same on the functional.

Reply:
That might be a bigger change. Given where this PR is right now, is there a reason why we can't use Tuples and keep BC?

Reply:
Yeah, let's go full BC for now.